sphinx/developers/external-readers.rst

```diff
@@ -0,0 +1,73 @@
+External file format readers
+============================
+
+Bio-Formats contains a file named :source:`readers.txt <components/formats-api/src/loci/formats/readers.txt>`
+which records every recognized reader class name. This file is used by the main reader class
+(:javadoc:`loci.formats.ImageReader <loci/formats/ImageReader.html>`) to associate a format reader
+with a particular file.
+
+For readers that are included in core Bio-Formats (i.e. the `ome/bioformats` repository on GitHub),
+a line containing just the fully qualified class name is added, e.g.:
+
+.. code-block::
+
+  loci.formats.in.OMETiffReader         # tif
+
+An optional comment can be added after the class name using `#`.
+
+Readers that are not in `ome/bioformats` and which are not included in Bio-Formats releases can be
+defined as "external" readers, e.g.:
+
+.. code-block::
+
+  loci.formats.in.ZarrReader[type=external]         # .zarr
+
+This allows Bio-Formats to recognize a reader if it was installed separately,
+but quietly ignore if it is not found.
+
+When to use an external reader
+------------------------------
+
+The OME team often encourages new readers to be proposed as "external" under any of the following conditions:
+
+- the newly introduced reader code is very large and/or complex
+- native libraries or closed-source dependencies are required
+- the reader covers an imaging domain which is unfamiliar to the OME team
+- test data cannot be provided to the OME team
+- the reader needs to be updated outside of the normal Bio-Formats release process
+
+When considering a new reader, we suggest discussing with the OME team first before making substantial changes.
+
+How to propose an external reader
+---------------------------------
+
+As indicated above, we suggest discussing with the OME team before investing significant time in a new reader.
+
+At minimum, open a new pull request on `ome/bioformats` that updates `readers.txt` as indicated above,
```

Comment:
Arguably this could be an issue as there is very little code involved and mostly discussions/agreement. But this is a detail.